### ISSUE TITLE
NullDereference is missing non-null case

### ISSUE THREAD

**Issue by emsod** [google/simplecfg#10]
This case produces a false positive:

if (x == null && y == null) {
   return null;
}
// If x is null then y is not null
z = (x != null) ? x.m() : y.m(); // false positive for y.m()

**Comment by llbit** [comment]
This false positive seems difficult to solve in the current architecture of the NullableDereference analyzer. It think a general solution to this would require some kind of constraint solver to know that `x == null` means that `y != null` after the first `if`-statement.

It might be possible to solve this problem for this specific case without a constraint solver, but it seems like a more general solution would need a constraint solver.

An idea for a constraint solving solution would be to build a set of constraints for each node in the control flow graph, for example the node following the else-part of the if-statement has the constraint `!(x == null && y == null)`, then the else-part of the conditional expression adds a constraint `x == null` to that set, leading to `y != null` being true.

**Comment by emsod** [comment]
Yes, I agree. This is a tricky case to solve and I think any constraint-like solution would require a lot of work. Due to this, I think there needs to be a way to filter out files we don't want to report findings for to avoid reporting the same false positive each time a file goes for review.
